src/logion/model/child.ts

```diff
@@ -65,21 +65,23 @@ export async function saveChildren<T extends Child>(parameters: Parameters<T>) {
     }
 }
 
-export function deleteChild<T extends Child>(childToDeleteIndex: number, children: T[], childrenToDelete: T[]) {
-    const childToDelete = children[childToDeleteIndex];
+export function deleteChild<T extends Child>(arrayIndex: number, children: T[], childrenToDelete: T[]) {
+    const childToDelete = children[arrayIndex];
     childrenToDelete.push(childToDelete)
-    children.splice(childToDeleteIndex, 1)
+    children.splice(arrayIndex, 1)
 }
 
-export function deleteIndexedChild<T extends Child & HasIndex>(childToDeleteIndex: number, children: T[], childrenToDelete: T[]) {
-    deleteChild(childToDeleteIndex, children, childrenToDelete)
-    reindexChildren(childToDeleteIndex, children)
+export function deleteIndexedChild<T extends Child & HasIndex>(arrayIndex: number, children: T[], childrenToDelete: T[]) {
```

Comment:
Did you check that callers are indeed passing in the array index and not the DB index?

Reply:
Yes. Basically all indexes at call sites are calculated from `array.findIndex()`.